### DISCUSSION TITLE
Alias block reference ID's with the source block

### DISCUSSION THREAD

**Discussion by ShreyasKallingal** [vslinko/obsidian-outliner#55] [Ideas]
**Is your feature request related to a problem? Please describe.**
When making block-level references, only the block ID is shown. This makes it difficult to make inline block references to specific parts.

![Screen Shot 2021-03-26 at 4 33 06 PM](https://user-images.githubusercontent.com/23715862/112702504-f393a700-8e50-11eb-87ec-416d45cd4c8a.png)

**Describe the solution you'd like**
 In other outliners like Roam and Remnote, the original block text is shown instead of the ID. Maybe it would be possible to create an alias for every block reference in a list such that the alias pulls from the original block text. 

For example:
![Screen Shot 2021-03-26 at 4 35 19 PM](https://user-images.githubusercontent.com/23715862/112702591-42d9d780-8e51-11eb-9eee-9bbef7d645e7.png)
becomes this in preview mode:
![Screen Shot 2021-03-26 at 4 35 38 PM](https://user-images.githubusercontent.com/23715862/112702605-4d946c80-8e51-11eb-951b-5069b6cba553.png)

**Describe alternatives you've considered**
I'm not sure about any alternatives to this.

**Comment by ShreyasKallingal** [comment]
I forgot about transclusions, my mistake! However, the problem remains that in-line transclusions aren't possible unlike other outliners. For example:
<img width="333" alt="Screen Shot 2021-03-26 at 8 07 48 PM" src="https://user-images.githubusercontent.com/23715862/112708381-f0a7af00-8e6e-11eb-88af-877f01ca656e.png">
turns into:
<img width="759" alt="Screen Shot 2021-03-26 at 8 07 56 PM" src="https://user-images.githubusercontent.com/23715862/112708387-f604f980-8e6e-11eb-9e8d-b14f7bd2bf61.png">

Maybe this is a CSS issue for block-level referencing? Please let me know what you think.

**Comment by vslinko** [comment]
@ShreyasKallingal Thank you for your contribution!

I think it's possible to implement a new type of reference like `((^6af37e))` that would work similar to Roam Research.
But I think it's better now to polish core functionality.

I'll keep this issue for future.

**Reply by MyEnthusiastic** [reply]
really hope this functionality. Now I have to link line by line, a lot of effort.

**Comment by vslinko** [comment]
I've converted the issue into discussions because it's very deep in my backlog and because it's possible to vote in discussions.

**Comment by mrpurplecloud** [comment]
First of all I'd like to vote this up as the lack of alias has been a pain when using block references. 

On the solution I would propose a quick and easy version: no additional syntax like ((^6af37e)), just extract the first or last N chars (configurable) of the line as the default alias for the block reference in context. This way users just need to develop their own conventions when composing text block, e.g always add a few words in the end as block alias. Don't know if it's technically feasible?
